auth/auth.go

```diff
@@ -781,3 +831,49 @@ func (auth *Authenticator) RegisterNewUser(username, email string) (User, error)
 
 	return user, err
 }
+
+// UpdateUserOIDCRolesChannels saves the user's current OIDC roles/channels, and resulting implicit roles/channels,
+// to the database.
+func (auth *Authenticator) UpdateUserOIDCRolesChannels(user_ User, roles base.Set, channels base.Set) error {
+	updateRolesChannelsCallback := func(currentPrincipal Principal) (updatedPrincipal Principal, err error) {
+		user, ok := currentPrincipal.(User)
+		if !ok {
+			return nil, base.ErrUpdateCancel
+		}
+
+		nextSeq := user.Sequence() + 1
+		user.SetSequence(nextSeq)
+
+		// TODO: these calls to rebuildRoles/rebuildChannels are somewhat expensive (require access/role-access queries)
+		// and we could hit this more than once if we're inside a CAS-loop.
+
+		newChans := user.OIDCChannels()
```

Comment:
This is named newChans, but looks like it's storing the previous/existing user channels, while the 'roles' and 'channels' parameters the newly updated roles/channels.  If I've got that right, can we name these 'existingChans'?

Reply:
It does start out as the user's previous OIDC channels, but then we add/remove the incoming channel set (to preserve the sequences of the pre-existing ones that don't get revoked). You're right on the naming though, it is a bit confusing, let me see if I can think of a better name.